packages/flutter/lib/src/services/process_text.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'system_channels.dart';
+
+/// A data structure describing text processing actions.
+@immutable
+class ProcessTextAction {
+  /// Creates text processing actions based on those returned by the engine.
+  const ProcessTextAction(this.id, this.label);
+
+  /// The action unique id.
+  final String id;
+
+  /// The action localized label.
+  final String label;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+
+    return other is ProcessTextAction &&
+      other.id == id &&
+      other.label == label;
+  }
+
+  @override
+  int get hashCode => Object.hash(id, label);
+}
+
+/// Determines how to interact with the text processing feature.
+abstract class ProcessTextService {
+  /// Returns a [Future] that resolves to a [List] of [ProcessTextAction]s
+  /// containing all text processing actions available.
+  Future<List<ProcessTextAction>> queryTextActions();
+
+  /// Returns a [Future] that resolves to a [String] when the text action
+  /// returns a transformed text or null when the text action did not return
+  /// a transformed text.
+  ///
+  /// The `id` parameter is the text action unique identifier returned by
+  /// [queryTextActions].
+  ///
+  /// The `text` parameter is the text to be processed.
+  ///
+  /// The `readOnly` parameter indicates that the transformed text, it if exists,
+  /// will be used as read-only.
+  /// See https://developer.android.com/reference/android/content/Intent#EXTRA_PROCESS_TEXT_READONLY.
+  Future<String?> processTextAction(String id, String text, bool readOnly);
```

Comment:
Is it purposeful that this is separated from the default service in the sense that the behavior here is intended to be customizable? Assuming so, a nit is maybe the link can be mentioned in the default service since this seems specific to Android?